src/com/ms/silverking/io/StreamParser.java

```diff
@@ -18,177 +16,182 @@
-        return parseLines(inStream, trimMode, CloseMode.close, null);
-    }
-    
-    public static List<String> parseLines(InputStream inStream, TrimMode trimMode, CloseMode closeMode, 
-                                        String regex) throws IOException {
-        BufferedReader        reader;
-        ArrayList<String>    lines;
-        String                line;
-        
-        reader = new BufferedReader( new InputStreamReader(inStream) );
-        try {
-            lines = new ArrayList<String>();
-            do {
-                line = reader.readLine();
-                if (line != null) {
-                    if (regex == null || line.matches(regex)) {
-                        if (trimMode == TrimMode.trim) {
-                            lines.add( line.trim() );
-                        } else {
-                            lines.add(line);
-                        }
-                    }
-                }
-            } while (line != null);
-        } finally {
-            if (closeMode == CloseMode.close) {
-                reader.close();
-            }
-        }
-        return lines;
-    }
-    
-    public static String parseLine(File file) throws IOException {
-        return parseLine(file, TrimMode.trim, CloseMode.close);
-    }
-    
-    public static String parseLine(File file, TrimMode trimMode, CloseMode closeMode) throws IOException {
-        return parseLine(new FileInputStream(file), trimMode, closeMode);
-    }
-    
-    public static String parseLine(InputStream inStream) throws IOException {
-        return parseLine(inStream, TrimMode.trim, CloseMode.close);
-    }
-    
-    public static String parseLine(InputStream inStream, TrimMode trimMode, CloseMode closeMode) throws IOException {
-        BufferedReader    reader;
-        
-        reader = new BufferedReader( new InputStreamReader(inStream) );
-        try {
+  public enum CloseMode {noClose, close}
+
+  ;
+
+  private static final char setDelimiterChar = ',';
+  private static final String setDelimiter = "" + setDelimiterChar;
+  private static final String mapChar = "=";
+
+  public static List<String> parseFileLines(File file, String regex) throws IOException {
+    return parseLines(new FileInputStream(file), regex);
+  }
+
+  public static List<String> parseFileLines(File file) throws IOException {
+    return parseLines(new FileInputStream(file));
+  }
+
+  public static List<String> parseGZFileLines(File file, TrimMode trimMode) throws IOException {
+    return parseLines(new GZIPInputStream(new FileInputStream(file)), trimMode);
+  }
+
+  public static List<String> parseFileLines(File file, TrimMode trimMode) throws IOException {
+    return parseLines(new FileInputStream(file), trimMode);
+  }
+
+  public static List<String> parseFileLines(String fileName) throws IOException {
+    return parseLines(new FileInputStream(fileName));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor call: *FileInputStream*. Currently, there are execution paths that do not contain closure statements, for example, when StreamParser.parseLines (in method parseLines) throws an exception. Either a) close the object returned by *FileInputStream()* in a try-finally block or b) close the resource by declaring the object returned by *FileInputStream()* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).